catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/resource/download/ReliableResourceDownloadManager.java

```diff
@@ -123,58 +97,73 @@ public ResourceResponse download(ResourceRequest resourceRequest, Metacard metac
             throw new DownloadException("Cannot download resource if request is null");
         }
 
-        try {
-            resourceResponse = retriever.retrieveResource();
-        } catch (ResourceNotFoundException | ResourceNotSupportedException | IOException e) {
-            throw new DownloadException("Cannot download resource", e);
+        if (downloaderConfig.isCacheEnabled()) {
+            Resource cachedResource = downloaderConfig.getResourceCache()
+                    .getValid(new CacheKey(metacard, resourceRequest).generateKey(), metacard);
+            if (cachedResource != null) {
+                resourceResponse = new ResourceResponseImpl(resourceRequest,
+                        resourceRequest.getProperties(),
+                        cachedResource);
+                LOGGER.debug("Successfully retrieved product from cache for metacard ID = {}",
+                        metacard.getId());
+            } else {
+                LOGGER.debug("Unable to get resource from cache. Have to retrieve it from source");
+            }
         }
 
-        resourceResponse.getProperties()
-                .put(Metacard.ID, metacard.getId());
-        // Sources do not create ResourceResponses with the original ResourceRequest, hence
-        // it is added here because it will be needed for caching
-        resourceResponse = new ResourceResponseImpl(resourceRequest,
-                resourceResponse.getProperties(),
-                resourceResponse.getResource());
-
-        // TODO - this should be before retrieveResource() but eventPublisher requires a
-        // resourceResponse and that resource response must have a resource request in it (to get
-        // USER property)
-        eventPublisher.postRetrievalStatus(resourceResponse,
-                ProductRetrievalStatus.STARTED,
-                metacard,
-                null,
-                0L,
-                downloadIdentifier);
-
-        AtomicBoolean downloadStarted = new AtomicBoolean(Boolean.FALSE);
-        ReliableResourceDownloader downloader = new ReliableResourceDownloader(downloaderConfig,
-                downloadStarted,
-                downloadIdentifier,
-                resourceResponse,
-                retriever);
-        resourceResponse = downloader.setupDownload(metacard, downloadStatusInfo);
-
-        // Start download in separate thread so can return ResourceResponse with
-        // ReliableResourceInputStream available for client to start reading from
-        executor.submit(downloader);
-
-        // Wait for download to get started before returning control to client
-        Stopwatch stopwatch = Stopwatch.createStarted();
-        while (!downloadStarted.get()) {
+        if (resourceResponse == null) {
             try {
-                Thread.sleep(10);
-            } catch (InterruptedException e) {
```

Comment:
Thanks for getting rid of this. I had added `Thread.currentThread().interrupt();` in the `catch` block as a bandaid to this bad practice, but I like what you did better. 

I might hold off on my changes until this gets into master. The conflicts wouldn't be worth it. I could just reapply my changes faster but redo-ing them on top of yours. How does that sound @lessarderic?